### PR TITLE
feat(android): Implement round and adaptive icon

### LINKED ISSUE #297: Round the corners of the android app icon
I use GitPoint on an Android device (but mainly iOS). I don’t like how the icon is squared (meaning not transparent anywhere). Most people prefer non-squared icons as well. So rounding the icon will make it look nicer.

### PR DESCRIPTION
**Pull request by machour**
In this PR:

- Updated the original squared icon to a squircle one
- Added `roundIcon` for devices supporting it since 7.1
- Added an adaptive icon in `mipmap-anydpi-v26`
- Updated compile/buildTools/target SDK versions

Since we're using `roundIcon` now, I was forced to duplicate the Android-O icon XML file,
 or else Android O kept using the PNGs provided in roundIcon.

Fixes #297 and #275

### CODE REVIEW

File: android/app/build.gradle [hunk 1]
```diff
@@ -92,13 +92,13 @@ def enableSeparateBuildPerCPUArchitecture = false
 def enableProguardInReleaseBuilds = false
 
 android {
-    compileSdkVersion 23
-    buildToolsVersion "23.0.1"
+    compileSdkVersion 26
+    buildToolsVersion '25.0.3'
```
Comment (lex111) [inline android/app/build.gradle]: Can the same up to 26 versions raise? I do not know for sure, just wondering.

Comment (machour) [reply android/app/build.gradle]: oopsy, updated

**Comment by machour** [comment]
Here's what the progressive icon looks like among other icons in Android O:

screen shot 2017-09-19 at 7 57 13 pm

@housseindjirdeh do we give it more margin to make it look like the rounded icon on https://gitpoint.co/? or do we do the opposite? 🤔

File: android/app/build.gradle [hunk 2]
```diff
@@ -92,13 +92,13 @@ def enableSeparateBuildPerCPUArchitecture = false
 def enableProguardInReleaseBuilds = false
 
 android {
-    compileSdkVersion 23
-    buildToolsVersion "23.0.1"
+    compileSdkVersion 26
+    buildToolsVersion '26.0.1'
```
Comment (lex111) [inline android/app/build.gradle]: I'll be pedantic, but let's change the quotes to double, as it used to be? :smile:

Comment (machour) [reply android/app/build.gradle]: ** curses Android Studio **

Comment (lex111) [reply android/app/build.gradle]: Heh, thank you :+1:

**Pull request merged**